espresso-server/app/src/test/java/io/appium/espressoserver/test/helpers/w3c/TickTest.java

```
@@ -0,0 +1,278 @@
+        tick.addAction(actionObjectOne);
+        tick.addAction(actionObjectTwo);
+
+        class ExtendedDummyW3CActionAdapter extends DummyW3CActionAdapter {
+            @Override
+            public void pointerMove(String sourceIdCalled,
+                                    PointerType pointerType,
+                                    long currentX, long currentY,
+                                    long x, long y,
+                                    Set<Integer> buttons,
+                                    KeyInputState globalKeyInputState) throws AppiumException {
+                assertEquals(pointerType, TOUCH);
+                assertEquals(currentX, 5L);
+                assertEquals(currentY, 6L);
+                assertEquals(x, 10L);
+                assertEquals(y, 20L);
+                assertTrue(buttons.contains(0));
+                assertTrue(buttons.contains(1));
+                assertTrue(globalKeyInputState.isShift());
+            }
+
+            @Override
+            public void keyDown(KeyEvent keyDownEvent) {
+            }
+        };
+
+        W3CActionAdapter baseW3CActionAdapter = new ExtendedDummyW3CActionAdapter();
+
+        List<Callable<Void>> callables = tick.dispatch(baseW3CActionAdapter, inputStateTable, tick.calculateTickDuration());
+        Executor executor = Executors.newFixedThreadPool(callables.size());
+        CompletionService<Void> completionService = new ExecutorCompletionService<>(executor);
+        for(Callable<Void> callable:callables) {
+            completionService.submit(callable);
+        }
+
+        int received = 0;
+        boolean errors = false;
+        while (received < callables.size() && !errors) {
+            Future<Void> resultFuture = completionService.take(); //blocks if none available
+            try {
+                resultFuture.get();
+                received ++;
+            }
+            catch(Exception e) {
```

Why exceptions should be caught in a test?

---

Good point, I could probably just not count the errors and let it throw the exception so that the test fails.